app/gui/view/graph-editor/src/component/node/growth_animation.rs

```diff
@@ -0,0 +1,99 @@
+//! Edited node growth/shrink animation implementation.
+//!
+//! When the user starts editing of the node - it smoothly growth in size to match the 1.0x zoom
+//! factor size. After editing is finished, the node smothly shrinks to its original size. This is
+//! implemented by using a separate `edited_node` camera that is moved in synchronization with
+//! `node_searcher` camera.
+
+use ensogl::prelude::*;
+
+use crate::GraphEditorModelWithNetwork;
+use crate::NodeId;
+use enso_frp as frp;
+use ensogl::animation;
+use ensogl::display::Scene;
+use ensogl::Animation;
+use ensogl::Easing;
+
+/// Initialize edited node growth/shrink animator. It would handle scene layer change for the edited
+/// node as well.
+pub fn initialize_edited_node_animator(
+    model: &GraphEditorModelWithNetwork,
+    frp: &crate::Frp,
+    scene: &Scene,
+) {
+    let network = &frp.network;
+    let out = &frp.output;
+    let searcher_cam = scene.layers.node_searcher.camera();
+    let edited_node_cam = scene.layers.edited_node.camera();
+    let main_cam = scene.layers.main.camera();
+
+    let growth_animation = Animation::new(network);
+    let animation_blending = Easing::new(network);
+
+    frp::extend! { network
+        let searcher_cam_frp = searcher_cam.frp();
+        let main_cam_frp = main_cam.frp();
+
+
+        // === Starting  node editing. ===
+
+        previous_edited_node <- out.node_editing_started.previous();
+        _eval <- all_with(&out.node_editing_started, &previous_edited_node, f!([model] (current, previous) {
+            move_node_to_main_layer(&model, *previous);
+            move_node_to_edited_node_layer(&model, *current);
+        }));
```

Comment:
I don't like these functions that take model as argument, as they clearly belong to model. This can be rewritten to:

```rust
_eval <- all_with(&previous_edited_node, &out.node_editing_started, f!((previous, current) {
    model.move_node_to_main_layer(previous);
    model.move_node_to_edited_node_layer(current);
}));
```

I know that model is in another module, but you can just extend it with a local trait in this module.

Reply:
It was written that way, but after moving to a separate module I changed methods to functions. Fixed it back now.